packages/curve-ui-kit/src/features/connect-wallet/ui/ConnectWalletIndicator.tsx

```diff
@@ -15,6 +15,13 @@ export const ConnectWalletIndicator = ({ sx }: { sx?: SxProps<Theme> }) => {
       sx={sx}
     />
   ) : (
-    <ConnectWalletButton onClick={() => connect()} loading={loading} sx={sx} />
+    <ConnectWalletButton
+      onClick={() => {
+        onConnect?.()
```

Comment:
I would emit the `onConnect` event after connect has been succesfully executed, just to be sure in case connect throws due to a wallet error.